syz-fuzzer/fuzzer.go

```diff
@@ -549,3 +555,12 @@ func parseOutputType(str string) OutputType {
 		return OutputNone
 	}
 }
+
+func collectMachineInfo() string {
```

Comment:
Please move this to a separate file. fuzzer.go is already quite large and this will grow.
Also needs to return an error.

Reply:
For the error handling part, because there are multiple machine information to collect (cpuinfo, kvm, etc), how should I handle it? Return an error if one of them fail? Currently I am thinking of recording as much information as possible, so if one of them fail we should still keep others. In this case I think returning a string make sense?

Reply:
But you already decided to abort the program in some case (where you used `log.Fatal`). So just return an error in places where you call `log.Fatal` and handle errors that you can handle locally.

The problem with silently throwing all errors away is that you also ignore all programming errors in your code. You can commit code with errors and nobody will notice because some errors are ignored and part of info is just not present in the result. Or the code may break later and nobody will notice again.

An error that is reasonable to handle locally is if we expect that some file may or may not be present and the error we get while opening it specially says that it's missing, then it's reasonable to ignore this error locally. But ignoring just _any_ error is wrong.

Reply:
And with such strategy you also won't be able to test this code.

Reply:
I changed it so that an error will be returned if it is not an `os.NotExist` error.